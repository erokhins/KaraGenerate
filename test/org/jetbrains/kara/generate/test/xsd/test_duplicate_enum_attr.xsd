<?xml version='1.0' encoding='utf-8' ?>
<xsd:schema xmlns='test' xmlns:xsd='http://www.w3.org/2001/XMLSchema' version="1.0" targetNamespace='test'>

    <xsd:element name="el1">
        <xsd:complexType>
            <xsd:attribute name="equalsAttr">
                <xsd:simpleType>
                    <xsd:restriction base="xsd:NMTOKEN">
                        <xsd:enumeration value="true"/>
                        <xsd:enumeration value="false"/>
                        <xsd:enumeration value="auto"/>
                    </xsd:restriction>
                </xsd:simpleType>
            </xsd:attribute>
            <xsd:attribute name="noEqualAttr">
                <xsd:simpleType>
                    <xsd:restriction base="xsd:NMTOKEN">
                        <xsd:enumeration value="true"/>
                        <xsd:enumeration value="false"/>
                        <xsd:enumeration value="auto"/>
                    </xsd:restriction>
                </xsd:simpleType>
            </xsd:attribute>
        </xsd:complexType>
    </xsd:element>


    <xsd:element name="el2">
        <xsd:complexType>
            <xsd:attribute name="equalsAttr">
                <xsd:simpleType>
                    <xsd:restriction base="xsd:NMTOKEN">
                        <xsd:enumeration value="true"/>
                        <xsd:enumeration value="false"/>
                        <xsd:enumeration value="auto"/>
                    </xsd:restriction>
                </xsd:simpleType>
            </xsd:attribute>
            <xsd:attribute name="noEqualAttr">
                <xsd:simpleType>
                    <xsd:restriction base="xsd:NMTOKEN">
                        <xsd:enumeration value="true"/>
                        <xsd:enumeration value="false"/>
                    </xsd:restriction>
                </xsd:simpleType>
            </xsd:attribute>
        </xsd:complexType>
    </xsd:element>

    <xsd:element name="el3">
        <xsd:complexType>
            <xsd:attribute name="noEqualAttr">
                <xsd:simpleType>
                    <xsd:restriction base="xsd:NMTOKEN">
                        <xsd:enumeration value="true"/>
                        <xsd:enumeration value="false"/>
                    </xsd:restriction>
                </xsd:simpleType>
            </xsd:attribute>
        </xsd:complexType>
    </xsd:element>

</xsd:schema>