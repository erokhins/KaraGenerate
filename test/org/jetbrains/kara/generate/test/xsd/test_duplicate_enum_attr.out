AttributeTypeDecl:
   AttributeTypeDecl = (equalsAttr, null, enumType)
   Values = [auto, false, true]

   AttributeTypeDecl = (noEqualAttr, el1, enumType)
   Values = [auto, false, true]

   AttributeTypeDecl = (noEqualAttr, el3, enumType)
   Values = [false, true]

AttributeDecl:
   AttributeDecl = (equalsAttr, null, null})
   AttributeTypeDecl = (equalsAttr, null)

   AttributeDecl = (noEqualAttr, el1, null})
   AttributeTypeDecl = (noEqualAttr, el1)

   AttributeDecl = (noEqualAttr, el3, null})
   AttributeTypeDecl = (noEqualAttr, el3)

AttributeGroupDecl:
ElementDecl:
   ElementDeclaration = (el1, false)
   ElementGroups = []
   NewAllowElements = []
   AttributeGroups = []
   NewAttributes =
      AttributeDecl = (equalsAttr, null, null})
      AttributeTypeDecl = (equalsAttr, null)

      AttributeDecl = (noEqualAttr, el1, null})
      AttributeTypeDecl = (noEqualAttr, el1)


   ElementDeclaration = (el2, false)
   ElementGroups = []
   NewAllowElements = []
   AttributeGroups = []
   NewAttributes =
      AttributeDecl = (equalsAttr, null, null})
      AttributeTypeDecl = (equalsAttr, null)

      AttributeDecl = (noEqualAttr, el3, null})
      AttributeTypeDecl = (noEqualAttr, el3)


   ElementDeclaration = (el3, false)
   ElementGroups = []
   NewAllowElements = []
   AttributeGroups = []
   NewAttributes =
      AttributeDecl = (noEqualAttr, el3, null})
      AttributeTypeDecl = (noEqualAttr, el3)


ElementGroup:
